meson.build

```
@@ -5,21 +5,76 @@
 # license: MIT
 #
 project('unity', 'c',
-    license: 'MIT',
-    meson_version: '>=0.37.0',
-    default_options: ['werror=true', 'c_std=c11'])
+  license: 'MIT',
+
+  # Set project version to value extracted from unity.h header
+  version: run_command(
+    [
+      find_program('python', 'python3'),
+      'auto' / 'extract_version.py',
+      'src' / 'unity.h'
+    ],
+    check: true
+  ).stdout().strip(),
+
+  # `meson.project_source_root()` introduced in 0.56.0
+  meson_version: '>=0.56.0',
+  default_options: [
+    'werror=true',
+    'c_std=c11'
+  ]
+)
+
+build_fixture = get_option('extension_fixture').enabled()
+build_memory = get_option('extension_memory').enabled()
+
+unity_src = []
+unity_inc = []
 
 subdir('src')
-unity_dep = declare_dependency(link_with: unity_lib, include_directories: unity_dir)
 
+if build_fixture
+  # Building the fixture extension implies building the memory
+  # extension.
+  build_memory = true
+  subdir('extras/fixture/src')
+endif
 
-# Get the generate_test_runner script relative to itself or the parent project if it is being used as a subproject
-# NOTE: This could be (and probably is) a complete hack - but I haven't yet been able to find a better way....
-if meson.is_subproject()
-gen_test_runner_path = find_program(meson.source_root() / 'subprojects/unity/auto/generate_test_runner.rb')
-else
-gen_test_runner_path = find_program('subprojects/unity/auto/generate_test_runner.rb')
+if build_memory
+  subdir('extras/memory/src')
 endif
 
-# Create a generator that we can access from the parent project
-gen_test_runner = generator(gen_test_runner_path, output: '@BASENAME@_Runner.c', arguments: ['@INPUT@', '@OUTPUT@'] )
+unity_lib = static_library(meson.project_name(),
+  sources: unity_src,
+  include_directories: unity_inc,
+  install: true
+)
+
+unity_dep = declare_dependency(
+  link_with: unity_lib,
+  include_directories: unity_inc
+)
+
+# Generate pkg-config file.
+pkg = import('pkgconfig')
+pkg.generate(unity_lib,
+  description: 'C Unit testing framework.'
+)
+
+# Create a generator that can be used by consumers of our build system to generate
+# test runners.
+gen_test_runner = generator(
+  find_program(meson.project_source_root() / 'auto' / 'generate_test_runner.rb'),
```

```suggestion
  find_program('auto/generate_test_runner.rb'),
```

find_program() will already search for local files relative to the current context meson.build file so you don't need to prepend an absolute path.

This also means you don't need to bump the minimum version just to use this.

---

Done. This simplifies things a lot.